libgenic/save.c

```diff
@@ -12,8 +12,9 @@
 #include <libgadget/utils.h>
 #include <libgadget/cosmology.h>
 #include <libgadget/walltime.h>
+#include <libgadget/utils/mymalloc.h>
 
-static void saveblock(BigFile * bf, void * baseptr, int ptype, char * bname, char * dtype, int items_per_particle, int64_t TotNumPart) {
+static void saveblock(BigFile * bf, void * baseptr, int ptype, char * bname, char * dtype, int items_per_particle, ptrdiff_t structsz, int64_t TotNumPart) {
```

Comment:
I usually call this term elsize (element size). I forgot where it came from, but it looks better than structsz.